### PR TITLE
Follow DART SE(2) convention

### LINKED ISSUE #455: Mismatch in logmap representation of SE(2) states
AIKIDO assumes [\theta, x, y] while DART assumes [x, y, theta]. Everytime the the skeleton is set to a position the coordinates are set incorrectly.

One clean solution would be to have the state representation to be equivalent to that of DART, OMPL: 
[x, y, theta].

Is there any particular reason we chose this representation?

### PR DESCRIPTION
**Pull request by aditya-vk**
Resolves #455.

***

**Before creating a pull request**

- [x] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [x] Set version target by selecting a milestone on the right side
- [x] Summarize this change in `CHANGELOG.md`
- [x] Add unit test(s) for this change

### CODE REVIEW

File: tests/constraint/test_SE2BoxConstraint.cpp [hunk 1]
```diff
@@ -21,15 +21,15 @@ class SE2BoxConstraintTests : public ::testing::Test
 {
 protected:
 #ifndef NDEBUG
-  static constexpr std::size_t NUM_A_TARGETS{5};
   static constexpr std::size_t NUM_X_TARGETS{5};
   static constexpr std::size_t NUM_Y_TARGETS{5};
+  static constexpr std::size_t NUM_A_TARGETS{5};
   static constexpr std::size_t NUM_SAMPLES{100};
-  static constexpr double DISTANCE_THRESHOLD{0.9};
+  static constexpr double DISTANCE_THRESHOLD{1.5};
```
Comment (aditya-vk) [inline tests/constraint/test_SE2BoxConstraint.cpp]: I am actually not sure what this threshold should be. It's a cool math problem though, if anyone figures it out, please let me know! The problem is:

Assume a unit square is discretized at some resolution `r`. Given `n` samples, I must be able to say that every grid point has at least one sample amongst the `n` that is at least as close as `x`. Find an upper bound on `x` given `n` and `r`.

Comment (brianhou) [reply tests/constraint/test_SE2BoxConstraint.cpp]: Why does this threshold need to change?

Comment (brianhou) [reply tests/constraint/test_SE2BoxConstraint.cpp]: Sorry, I guess my question is really why do both `NUM_SAMPLES` and `DISTANCE_THRESHOLD` need to change? I'd rather keep `NUM_SAMPLES` the same and just increase the threshold. (Again, this change is just to ensure that non-debug builds are as fast as debug builds.)

**Comment by jslee02** [comment]
CI fails due to style inconsistency.

**Comment by aditya-vk** [comment]
CI tests all seemed to have passed?

File: tests/constraint/test_SE2BoxConstraint.cpp (hunk 1 above)
Comment (aditya-vk) [reply tests/constraint/test_SE2BoxConstraint.cpp]: I increased the number just for it to be consistent with the non-debug style i.e number of samples in the grid = number of samples used for coverage tests. It shouldn't change the build speed much :)

The changes made in #418 for faster tests should hold.

**Comment by jslee02** [comment]
Oh, sorry. I was confused with #457 :sweat_smile:

**Comment by aditya-vk** [comment]
Subtle hint from @jslee02 that I should complete #457 soon? yessir!

**Pull request merged**